tests/e2e/velero_yaml_suite_test.go

```diff
@@ -0,0 +1,44 @@
+package e2e
+
+import (
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+type dpaTemplateParams struct {
+	DpaName         string
+	BslRegion       string
+	ClusterProfile  string
+	Provider        string
+	CredentialsName string
+	BucketName      string
+	Prefix          string
+}
+
+var _ = AfterEach(func() {
+	// Delete DPA once the test has finished
+	err := vel.Delete()
```

Comment:
So this is actually not cleaning up the DPA CR that you created. By moving to this template creation workflow the existing functions won't be useful here. `vel.Delete()` depends on the DPA CR struct existing in memory on the `vel` resource.

You could write a new function:
```
func DeleteDPAFromYaml() error {
    // Take in the DPA template and issue a client.Delete() based on the template name
}
```